### PR TITLE
Remove useless pylint error suppression directives

### PR DESCRIPTION
**Pull request by adferrand**
As pylint is evolving, it improves its accuracy, and several pylint error suppression (`# pylint: disable=ERROR) added in certbot codebase months or years ago are not needed anymore to make it happy.

There is a (disabled by default) pylint error to detect the useless suppressions (pylint-ception: `useless-suppression`). It is not working perfectly (it has also false-positives ...) but it is a good start to clean the codebase.

This PR removes several of these useless suppressions as detected by the current pylint version we use.

### CODE REVIEW

**Comment by adferrand** [comment]
For the record, this requires to enable the `useless-suppression` error, run pylint, read the output, fix the code, and repeat. Since this is extremely boring, I made a script that does that for me.

```python
#!/usr/bin/env python3

import re

from pylint.lint import Run

SOURCE_PATHS = [   
    'acme/acme',
    'certbot/certbot',
    'certbot-apache/certbot_apache',
    'certbot-compatibility-test/certbot_compatibility_test',
    'certbot-dns-cloudflare/certbot_dns_cloudflare',
    'certbot-dns-cloudxns/certbot_dns_cloudxns',
    'certbot-dns-digitalocean/certbot_dns_digitalocean',
    'certbot-dns-dnsimple/certbot_dns_dnsimple',
    'certbot-dns-dnsmadeeasy/certbot_dns_dnsmadeeasy',
    'certbot-dns-gehirn/certbot_dns_gehirn',
    'certbot-dns-google/certbot_dns_google',
    'certbot-dns-linode/certbot_dns_linode',
    'certbot-dns-luadns/certbot_dns_luadns',
    'certbot-dns-nsone/certbot_dns_nsone',
    'certbot-dns-ovh/certbot_dns_ovh',
    'certbot-dns-rfc2136/certbot_dns_rfc2136',
    'certbot-dns-route53/certbot_dns_route53',
    'certbot-dns-sakuracloud/certbot_dns_sakuracloud',
    'certbot-nginx/certbot_nginx',
    'letshelp-certbot/letshelp_certbot',
    'tests/lock_test.py',
]

PYLINT_DISABLE_DIRECTIVE_PATTERN = re.compile(r'(?:\||#)\s*pylint\s*:\s*disable\s*=\s*(.*)')
USELESS_SUPPRESSION_MESSAGE_PATTERN = re.compile(r'Useless suppression of \'(.*)\'')

def update_pylint_disable_directive(path, line_idx, to_remove):
    with open(path, 'r') as file_h:
        lines = file_h.read().splitlines()

    line = lines[line_idx - 1]
    match = PYLINT_DISABLE_DIRECTIVE_PATTERN.search(line)
    directives = [directive.strip() for directive in match.group(1).split(',')]

    if (to_remove in ('missing-module-docstring', 'missing-class-docstring', 'missing-function-docstring')
            and 'missing-docstring' in directives):
        directives.extend([directive for directive in [
            'missing-module-docstring', 'missing-class-docstring', 'missing-function-docstring']
            if directive != to_remove])
        to_remove = 'missing-docstring'
    directives = [directive for directive in directives if directive != to_remove]

    line = PYLINT_DISABLE_DIRECTIVE_PATTERN.sub('', line)
    if directives:
        line = line + '# pylint: disable={0}'.format(','.join(directives))
    if not line.strip():
        lines.pop(line_idx - 1)
    else:
        lines[line_idx - 1] = line.rstrip()

    with open(path, 'w') as file_h:
        file_h.writelines(['{0}\n'.format(line) for line in lines])

run = Run(['--rcfile=.pylintrc', '--enable=useless-suppression', '--output-format=json', *SOURCE_PATHS], do_exit=False)

for message_obj in run.linter.reporter.messages:
    try:
        to_remove = USELESS_SUPPRESSION_MESSAGE_PATTERN.search(message_obj['message']).group(1)
        line_idx = message_obj['line']
        path = message_obj['path']

        print('Removing directive "{0}" on {1}:L{2}.'.format(to_remove, path, line_idx))

        update_pylint_disable_directive(path, line_idx, to_remove)
    except:
        pass
```

To be run inside a virtual environment created with `tools/venv3.py`. To be run several time until there is no `useless-suppression` error anymore, then fix back the 3 or 4 lint errors that have been added because of `useless-suppression` false positives.

File: acme/acme/crypto_util.py [hunk 1]
```diff
@@ -11,10 +11,10 @@
 from OpenSSL import SSL  # type: ignore # https://github.com/python/typeshed/issues/2052
 
 from acme import errors
-from acme.magic_typing import Callable  # pylint: disable=unused-import, no-name-in-module
-from acme.magic_typing import Optional  # pylint: disable=unused-import, no-name-in-module
-from acme.magic_typing import Tuple  # pylint: disable=unused-import, no-name-in-module
-from acme.magic_typing import Union  # pylint: disable=unused-import, no-name-in-module
+from acme.magic_typing import Callable
+from acme.magic_typing import Optional  # pylint: disable=unused-import
```
Comment (bmw) [inline acme/acme/crypto_util.py]: While we're here, can we delete this line?

File: acme/acme/standalone.py [hunk 2]
```diff
@@ -5,18 +5,17 @@
...
 from acme import challenges
 from acme import crypto_util
-from acme.magic_typing import List  # pylint: disable=unused-import, no-name-in-module
+from acme.magic_typing import List
 
 logger = logging.getLogger(__name__)
 
-# six.moves.* | pylint: disable=no-member,attribute-defined-outside-init
-# pylint: disable=no-init
+# six.moves.*
```
Comment (bmw) [inline acme/acme/standalone.py]: I think we can just delete this comment now.

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py [hunk 3]
```diff
@@ -148,7 +148,7 @@ def add_txt_record(self, domain, record_name, record_content, record_ttl):
                 },
             ]
 
-        changes = self.dns.changes()  # changes | pylint: disable=no-member
+        changes = self.dns.changes()  # changes
```
Comment (bmw) [inline certbot-dns-google/certbot_dns_google/_internal/dns_google.py]: I think we can just delete this comment.

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py [hunk 4]
```diff
@@ -213,7 +213,7 @@ def del_txt_record(self, domain, record_name, record_content, record_ttl):
...
```
Comment (bmw) [inline certbot-dns-google/certbot_dns_google/_internal/dns_google.py]: I think we can just delete this comment.

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py [hunk 5]
```diff
@@ -264,7 +264,7 @@ def _find_managed_zone_id(self, domain):
 
         zone_dns_name_guesses = dns_common.base_domain_name_guesses(domain)
 
-        mz = self.dns.managedZones()  # managedZones | pylint: disable=no-member
+        mz = self.dns.managedZones()  # managedZones
```
Comment (bmw) [inline certbot-dns-google/certbot_dns_google/_internal/dns_google.py]: I think we can just delete this comment.

File: certbot/certbot/_internal/plugins/disco.py [hunk 6]
```diff
@@ -206,7 +204,7 @@ def find_all(cls):
             plugin_ep = PluginEntryPoint(entry_point)
             assert plugin_ep.name not in plugins, (
                 "PREFIX_FREE_DISTRIBUTIONS messed up")
-            # providedBy | pylint: disable=no-member
+            # providedBy
```
Comment (bmw) [inline certbot/certbot/_internal/plugins/disco.py]: I think we can just delete this comment.

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py (hunk 5 above)
Comment (adferrand) [reply certbot-dns-google/certbot_dns_google/_internal/dns_google.py]: Done.

File: acme/acme/standalone.py (hunk 2 above)
Comment (adferrand) [reply acme/acme/standalone.py]: Done.

File: acme/acme/crypto_util.py (hunk 1 above)
Comment (adferrand) [reply acme/acme/crypto_util.py]: Done.

**Comment by ohemorange** [comment]
I'm now hitting these pylint errors locally. What do I need to update to not do that? Reinstalling the venv didn't work, I'm getting it on both py2.7 and py3.

**Comment by bmw** [comment]
Sorry for the trouble! Are these the only errors you're getting locallly?
```
************* Module acme.challenges
acme/acme/challenges.py:57:15: E1101: Instance of 'UnrecognizedChallenge' has no 'jobj' member (no-member)
************* Module acme.jws
acme/acme/jws.py:28:16: E1101: Class 'Signature' has no '_orig_slots' member (no-member)
```
If so, this is fixed by https://github.com/certbot/certbot/pull/7780. It appears it's a bug in pylint or its dependencies where it's only complaining about these lines in specific versions of Python 3.5 and 3.6.

**Comment by ohemorange** [comment]
Nice catch! That was it (I use 3.6.9), thanks for the quick fix.

**Pull request merged**